### PR TITLE
correcting docstrings in asg.py

### PR DESCRIPTION
**Pull request by joshuaroot**
- correcting docstring examples for
  - asg.filter.launch-config
  - asg.filters.image

### CODE REVIEW

File: c7n/resources/asg.py [hunk 1]
```diff
@@ -647,12 +647,12 @@ class VpcIdFilter(ValueFilter):
...
-                resource: asg
-                filters:
-                  - type: vpc-id
-                    value: vpc-12ab34cd
+        policies:
+          - name: asg-vpc-xyz
+            resource: asg
+            filters:
+              - type: vpc-id
+                value: vpc-12ab34cd
```
Comment (kapilt) [inline c7n/resources/asg.py]: That’s not a valid filter

File: c7n/resources/asg.py [hunk 2]
```diff
@@ -157,13 +157,13 @@ class LaunchConfigFilter(ValueFilter, LaunchConfigFilterBase):
...
-                filters:
-                  - type: launch-config
-                    key: AssociatePublicIpAddress
-                    value: true
+        policies:
+          - name: asg-launch-config-using-image
+            resource: asg
+            filters:
+              - type: launch-config
+                key: ImageId
```
Comment (kapilt) [inline c7n/resources/asg.py]: The non public up is probably a more useful example

File: c7n/resources/asg.py (hunk 1 above)
Comment (joshuaroot) [reply c7n/resources/asg.py]: how so?

File: c7n/resources/asg.py (hunk 2 above)
Comment (joshuaroot) [reply c7n/resources/asg.py]: i changed this one as i was unable to get the `AssociatePublicIpAddress` to actually give me a valid return. when i look at what the `LaunchConfigFilterBase` pulls out, i do not see that value in the response.

File: c7n/resources/asg.py (hunk 1 above)
Comment (kapilt) [reply c7n/resources/asg.py]: oh.. sorry i guess it is, it just seems odd to have a dedicated id filter when its commonly referenced via value filter as an attribute, this is one thats worth deprecating for a `vpc` filter that can look at additional vpc attributes (tags, etc).

Comment (joshuaroot) [reply c7n/resources/asg.py]: yeah, agreed. it actually says as much in the docstring.

Comment (joshuaroot) [reply c7n/resources/asg.py]: actually, the vpc-id does not show up in the `describe_autoscaling_groups` generic results.. the `VPCZoneIdentifier` gives a subnet id which is what the filter keys on to find the vpc.

Comment (kapilt) [reply c7n/resources/asg.py]: right, but doesn't really change that this should be noted in the docs as deprecated, and we should have a vpc filter here that can more usefully filter asgs based on other vpc attributes.

File: c7n/resources/asg.py (hunk 2 above)
Comment (kapilt) [reply c7n/resources/asg.py]: AssociatePublicIpAddress is an attribute of the launch config per aws docs.

Comment (joshuaroot) [reply c7n/resources/asg.py]: bah, yes, ok, so it is a valid `key` if it is specified in the launch-config... if you don't select `Assign a public IP address to every instance.` that key doesn't seem to be attached... so it's either `true` or `absent` for a valid result.

File: c7n/resources/asg.py (hunk 1 above)
Comment (jhnlsn) [reply c7n/resources/asg.py]: @kapilt wouldn't we need to have an alternative before we denote the functionality as deprecated?  I don't think we have the vpc filter functionality yet.  This PR is just updating incorrect documentation.

Comment (kapilt) [reply c7n/resources/asg.py]: thats fair, in this context though its not even correcting (re vpc-id), its just reformatting it.

Comment (kapilt) [reply c7n/resources/asg.py]: in fact this whole pr is an exercise in reformatting not correctness.

**Pull request merged**